internal/pkg/vm/internal/vmcontext/vmcontex2.go

```diff
@@ -295,12 +363,13 @@ func (vm *VM) applyMessage(msg *types.UnsignedMessage, onChainMsgSize uint32, mi
 
```

Comment:
I'm not high context enough to start making incorrectness claims, but it would be worth taking a look at consensus/validation.go to make sure we are doing all the necessary checks here.  It looks like this covers most but not all of the checks we used to do on message validation.  Missing some checks allowing high nonce, negative value, block gas limit.  

No idea if this is more up to date, just flagging.

Reply:
This is nearly verbatim the spec up to dec 12th. 
There are some new changes now, so this needs to get revised. 

FWIW, any check not here and not in the spec I think its expected to happen BEFORE the messages are selected for execution. What I think of as "Message validation" happens way before here.

Reply:
Yeah that naming suits me more too but consensus/validation.go contains the old version of these rules 🤷‍♂